boreal/src/bitmaps.rs

```diff
@@ -0,0 +1,166 @@
+use std::ops::BitOrAssign;
+
+#[derive(Debug, Clone, Copy, Default)]
+/// A bitmap with 256 bits
+pub struct Bitmap {
+    low: u128,
+    high: u128,
+}
+
+impl Bitmap {
+    const HALF: u8 = 128;
+
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    fn mask(bit: u8) -> u128 {
+        1u128 << (bit & 127)
+    }
+
+    fn get_half(&self, bit: u8) -> u128 {
+        if bit < Self::HALF {
+            self.low
+        } else {
+            self.high
+        }
+    }
+
+    fn get_half_mut(&mut self, bit: u8) -> &mut u128 {
+        if bit < Self::HALF {
+            &mut self.low
+        } else {
+            &mut self.high
+        }
+    }
+
+    #[must_use]
+    #[inline(always)]
+    pub fn get(&self, bit: u8) -> bool {
+        let mask = Self::mask(bit);
+        let half = self.get_half(bit);
+        half & mask != 0
+    }
+
+    #[inline(always)]
+    pub fn set(&mut self, bit: u8, value: bool) {
```

Comment:
This is actually only called with `value=true`, except in the iterator implementation. but with my iter implementation suggestion, it is actually no longer used at all, so it can be simplified and the value removed: there is no api to set a bit to false (this is fine for our use of the bitmap)